strongbox-aql/src/main/java/org/carlspring/strongbox/services/impl/AqlSearchServiceImpl.java

```diff
@@ -59,7 +59,7 @@ public SearchResults search(Selector<ArtifactEntry> selector)
                                                                            artifactEntry.getRepositoryId(),
                                                                            artifactEntry.getArtifactPath());
 
-            Repository repository = repositoryPath.getRepository();
+            RepositoryData repository = repositoryPath.getRepository();
```

Comment:
Why do we need to rename `Repository` to `RepositoryData`? I'm not quite sure I like the proposed new name.

To be honest, I'm not the biggest fan ot the whole `Repository` + `MutableRepository` concept, but okay.

Reply:
> To be honest, I'm not the biggest fan ot the whole Repository + MutableRepository concept, but okay.

What else do you propose, @carlspring ?

Reply:
Unfortunately, I don't have a better proposal. I understand it was implemented like this for better beformance and synchronization. I personally, think it's a bit hard to work with it this way, maybe it's just me. It reminds me of the days of RMI...

Reply:
yeah, I agree that this is not very comfortable to work with it, but there is an option to use `POJONATOR` :laughing: